platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptEvaluator.kt

```diff
@@ -306,6 +301,35 @@ class StandardKotlinScriptEvaluator(
             ?: executionEngine
     }
 
+    private
+    fun unitOWork(
+        programId: ProgramId,
+        compilationClassPath: ClassPath,
+        accessorsClassPath: ClassPath,
+        initializer: (File) -> Unit
+    ): BuildScriptCompileUnitOfWork {
+        val displayName = "Kotlin DSL script compilation (${programId.templateId})"
+        val inputs = BuildScriptCompileInputs { visitor ->
+            visitor.visitInputProperty(JVM_TARGET) { programId.compilerOptions.jvmTarget.majorVersion }
+            visitor.visitInputProperty(ALL_WARNINGS_AS_ERRORS) { programId.compilerOptions.allWarningsAsErrors }
+            visitor.visitInputProperty(SKIP_METADATA_VERSION_CHECK) { programId.compilerOptions.skipMetadataVersionCheck }
+            visitor.visitInputProperty(TEMPLATE_ID) { programId.templateId }
+            visitor.visitInputProperty(SOURCE_HASH) { programId.sourceHash }
+            visitor.visitInputProperty(COMPILATION_CLASS_PATH) { classpathHasher.hash(compilationClassPath) }
+            visitor.visitInputProperty(ACCESSORS_CLASS_PATH) { classpathHasher.hash(accessorsClassPath) }
+        }
```

Comment:
❓ Is it necessary to introduce `BuildScriptCompileInputs`? Why not leave `visitInputs()` abstract in `BuildScriptCompileUnitOfWork` and implement it in place instead?